pkg/sync/http/http_sync.go

```diff
@@ -0,0 +1,140 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1" //nolint:gosec
+	"encoding/base64"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/open-feature/flagd/pkg/sync"
+
+	"github.com/open-feature/flagd/pkg/logger"
+)
+
+type Sync struct {
+	URI          string
+	Client       Client
+	Cron         Cron
+	BearerToken  string
+	LastBodySHA  string
+	Logger       *logger.Logger
+	ProviderArgs sync.ProviderArgs
+}
+
+// Client defines the behaviour required of a http client
+type Client interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Cron defines the behaviour required of a cron
+type Cron interface {
+	AddFunc(spec string, cmd func()) error
+	Start()
+	Stop()
+}
+
+func (hs *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	// Initial fetch
+	fetch, err := hs.Fetch(ctx)
+	if err != nil {
+		hs.Logger.Error(err.Error())
+		return err
+	}
+
+	dataSync <- sync.DataSync{FlagData: fetch, Source: hs.URI}
+
+	_ = hs.Cron.AddFunc("*/5 * * * *", func() {
```

Comment:
This ought to be configurable

Reply:
I will create a follow up issue :)

Reply:
Created issue #293 . IMO, this is a "good first issue" candidate.